resources/lib/tubecast/youtube/app.py

```diff
@@ -430,30 +451,53 @@ def __monitor_volume(self):
 class YoutubeListener(threading.Thread):
 
     def __init__(self, app, ssdp=True):
-        threading.Thread.__init__(self)
-        self.app = app
+        super(YoutubeListener, self).__init__(name="YoutubeListener")
+        self.app = app  # type: YoutubeCastV1
         self.stop = False
         self.ssdp = ssdp
+        self.r = None  # type: Optional[requests.Response]
+
+    def __read_cmd_lines(self, url):  # type: (str) -> Iterator[str]
```

Comment:
I added this new method to handle the `ChunkedEncodingError` which is raised when we close the socket.
This is a generator function which essentially just mimics the `iter_lines` iterator.

Note that even though I'm using some force to cancel the request it still gets the chance to clean up after itself.